processor/spanmetricsprocessor/processor.go

```diff
@@ -308,6 +328,12 @@ func (p *processorImp) collectLatencyMetrics(ilm pmetric.ScopeMetrics) error {
 	dps.EnsureCapacity(len(p.histograms))
 	timestamp := pcommon.NewTimestampFromTime(time.Now())
 	for key, hist := range p.histograms {
+		// If the key is not in the cache, simply don't publish it because it's likely
+		// the metric has expired (or the cache size is too small).
+		dimensions, ok := p.metricKeyToDimensions.Get(key)
+		if !ok {
+			continue
+		}
```

Comment:
Isn't this silently ignoring the error? Wouldn't we still want to log the error message that the key was not found? Otherwise, how would one know why some metrics are missing from the collection?

Reply:
Yup, good point, thanks for pointing this out. I've added a warning log message and removed the ineffectual comment because I think the log message should document the cause of the issue well enough. Done in: [8c66291](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/17307/commits/8c662917005db89ca868339d841fbaf88e52bb3c).